src/bam/read_orientation.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Johannes Köster.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use strum_macros::AsRefStr;
+
+use crate::bam;
+
+/// Representation of read orientation
+/// (e.g. F1R2 means that the forward read comes first on the reference contig,
+/// followed by the reverse read, on the same contig).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, AsRefStr)]
+#[cfg_attr(feature = "serde_feature", derive(Serialize, Deserialize))]
+pub(crate) enum ReadOrientation {
+    F1R2,
+    F2R1,
+    R1F2,
+    R2F1,
+    F1F2,
+    R1R2,
+    F2F1,
+    R2R1,
+    None,
+}
+
+impl<'a> From<&'a bam::Record> for ReadOrientation {
+    /// Infer read orientation from given BAM record. Returns `ReadOrientation::None` if record
+    /// is not properly paired, mates are not mapping to the same contig, or mates start at the
+    /// same position.
+    fn from(record: &bam::Record) -> Self {
+        if record.is_paired() && record.is_proper_pair() && record.tid() == record.mtid() {
+            if record.pos() == record.mpos() {
```

Comment:
Why? Any adapter trimmed read has identical coordinates, but you can still infer the orientation?

Reply:
The rationale was: if both reads start at the same position, there is no read that comes "first". How would you infer which one comes first in such a case?

Reply:
Ah, I assumed the first_in_template was responsible for that (I didn't look very closely at the match table logic)